lib/matplotlib/tests/test_ticker.py

```diff
@@ -452,23 +452,111 @@ def test_percentformatter():
+
+    # Case B/ With kwarg `places=0`
+    B_wo_unit_w_sep = ((-1234.56789, u'-1 k'),
+                       (-1.23456789, u'-1'),
+                       (-0.123456789, u'-123 m'),
+                       (-0.00123456789, u'-1 m'),
+                       (-0.0, u'0'),
+                       (-0, u'0'),
+                       (0, u'0'),
+                       (1.23456789e-6, u'1 \u03bc'),
+                       (0.123456789, u'123 m'),
+                       (1, u'1'),
+                       (1.23456789, u'1'),
+                       (999.9, u'999'),
+                       (1000, u'1 k'),
+                       (1001, u'1 k'),
+                       (100001, u'100 k'),
+                       (987654.321, u'987 k'))
+
+    # Case C/ With kwarg `places=2`
+    C_wo_unit_w_sep = ((-1234.56789, u'-1.23 k'),
+                       (-1.23456789, u'-1.23'),
+                       (-0.123456789, u'-123.46 m'),
+                       (-0.00123456789, u'-1.23 m'),
+                       (-0.0, u'0.00'),
+                       (-0, u'0.00'),
+                       (0, u'0.00'),
+                       (1.23456789e-6, u'1.23 \u03bc'),
+                       (0.123456789, u'123.46 m'),
+                       (1, u'1.00'),
+                       (1.23456789, u'1.23'),
+                       (999.9, u'999.90'),
+                       (1000, u'1.00 k'),
+                       (1001, u'1.00 k'),
+                       (100001, u'100.00 k'),
+                       (987654.321, u'987.65 k'))
+
+    for ref_cases, places in ((A_wo_unit_w_sep, None),
+                              (B_wo_unit_w_sep, 0),
+                              (C_wo_unit_w_sep, 2)):
+
+        """ Test an EngFormatter instance w/o unit and w/ space separator
+        (default)
+        """
+        eng_formatter = mticker.EngFormatter(places=places)
+        for value, expected_string in ref_cases:
+            nose.tools.assert_equal(eng_formatter(value), expected_string)
+
+        """ Test an EngFormatter instance w/ unit and w/ space separator
+        """
+        eng_formatter = mticker.EngFormatter(unit=UNIT, places=places)
+        # Append a unit symbol to the reference cases
+        # Beware of the case of values in [1, 1000), where there is no prefix!
+        test_cases = ((val, u"{s:s} {u:s}".format(s=s, u=UNIT))
+                      if s[-1] in DIGITS  # i.e. if no prefix is present
+                      else (val, u"{s:s}{u:s}".format(s=s, u=UNIT))
+                      for val, s in ref_cases)
+        for value, expected_string in test_cases:
+            nose.tools.assert_equal(eng_formatter(value), expected_string)
+
+        """ Test an EngFormatter instance w/o unit and w/o space separator
+        """
+        eng_formatter = mticker.EngFormatter(places=places, space_sep=False)
+        # Remove the space separator from the reference cases
+        test_cases = ((val, u"{s:s}".format(s=s.replace(" ", "")))
```

Comment:
I simply copy-pasted from one of the previous similar case and didn't pay more attention, but you are right: it's clearer without the string format. Removed by commit  8d086e0.